### ISSUE TITLE
CA1508 False positive for "'x != null' is always 'true'"

### ISSUE THREAD

**Issue by bartdesmet** [dotnet/roslyn-analyzers#6164]
### Steps To Reproduce

```csharp
#define REPRO

public static void Bar(ConsoleColor c)
{
    var color = default(string?);

    switch (c)
    {
        case ConsoleColor.Red:
        case ConsoleColor.DarkRed:
            color = "red";
            break;
        case ConsoleColor.Blue:
        case ConsoleColor.DarkBlue:
            color = "blue";
            break;
        default:
#if !REPRO
            color = null;
#endif
            break;
    }

    if (color != null)
#if REPRO
//      ~~~~~
//      CA1508: 'color != null' is always 'true'
#endif
    {
        // stuff
    }
}
```

### Expected behavior

No warning.

### Actual behavior

Issues warning.

## Additional context

Note that Roslyn also reports nullable analysis info in the tooltip stating "'color' may be null here.", so the analyzer and compiler have different opinions here.
